support/cas-server-support-ehcache3-ticket-registry/src/main/java/org/apereo/cas/config/Ehcache3TicketRegistryConfiguration.java

```diff
@@ -0,0 +1,205 @@
+import org.apereo.cas.ticket.Ticket;
+import org.apereo.cas.ticket.TicketCatalog;
+import org.apereo.cas.ticket.TicketDefinition;
+import org.apereo.cas.ticket.registry.EhCache3TicketRegistry;
+import org.apereo.cas.ticket.registry.TicketRegistry;
+import org.apereo.cas.util.CoreTicketUtils;
+
+import lombok.extern.slf4j.Slf4j;
+import lombok.val;
+import org.apache.commons.lang3.StringUtils;
+
+import org.ehcache.clustered.client.config.builders.ClusteredResourcePoolBuilder;
+import org.ehcache.clustered.client.config.builders.ClusteredStoreConfigurationBuilder;
+import org.ehcache.clustered.client.config.builders.ClusteringServiceConfigurationBuilder;
+import org.ehcache.clustered.client.config.builders.TimeoutsBuilder;
+import org.ehcache.clustered.common.Consistency;
+import org.ehcache.config.CacheConfiguration;
+import org.ehcache.config.builders.CacheConfigurationBuilder;
+import org.ehcache.config.builders.CacheEventListenerConfigurationBuilder;
+import org.ehcache.config.builders.ExpiryPolicyBuilder;
+import org.ehcache.config.builders.ResourcePoolsBuilder;
+import org.ehcache.config.units.MemoryUnit;
+import org.ehcache.core.config.DefaultConfiguration;
+import org.ehcache.event.CacheEvent;
+import org.ehcache.event.CacheEventListener;
+import org.ehcache.event.EventType;
+import org.ehcache.impl.config.persistence.DefaultPersistenceConfiguration;
+import org.ehcache.jsr107.Eh107Configuration;
+import org.ehcache.jsr107.EhcacheCachingProvider;
+import org.ehcache.jsr107.config.ConfigurationElementState;
+import org.ehcache.jsr107.config.Jsr107Configuration;
+import org.ehcache.spi.service.ServiceCreationConfiguration;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.cache.jcache.JCacheCacheManager;
+import org.springframework.cloud.context.config.annotation.RefreshScope;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+import javax.cache.Caching;
+
+import java.io.File;
+import java.net.URI;
+import java.time.Duration;
+import java.util.HashMap;
+
+/**
+ * This is {@link Ehcache3TicketRegistryConfiguration}.
+ *
+ * @author Hal Deadman
+ * @since 6.2.0
+ */
+@Configuration(value = "ehcache3TicketRegistryConfiguration", proxyBeanMethods = false)
+@EnableConfigurationProperties(CasConfigurationProperties.class)
+@ConditionalOnProperty(prefix = "cas.ticket.registry.ehcache3", name = "enabled", havingValue = "true", matchIfMissing = true)
+@Slf4j
+public class Ehcache3TicketRegistryConfiguration {
+
+    @Autowired
+    private CasConfigurationProperties casProperties;
+
+    @Bean
+    @ConditionalOnMissingBean
```

Comment:
Name the conditionals by the bean name please, and not rely on type itself to avoid conflicts.